src/codeu/chat/client/commandline/Chat.java

```diff
@@ -136,8 +148,11 @@ public void invoke(Scanner args) {
     //
     panel.register("u-add", new Panel.Command() {
       @Override
-      public void invoke(Scanner args) {
-        final String name = args.hasNext() ? args.nextLine().trim() : "";
+      public void invoke(List<String> args) {
+        //TODO: possible place of error, not sure how to test username creation
+        // what's in the List being passed in? Is it already tokenized? The scanner code took the entire line?
```

Comment:
Hmm, I don't think I understand. Are you not able to change line 81 to `if (panels.peek().handleCommand(command, tokenizer)) {` and remove the entire loop and `args` construction above it?